src/tests/test_system.py

```
@@ -0,0 +1,276 @@
+        print(f"✓ Strategy validation working")
+
+def run_tests():
+    """Run all tests with detailed output."""
+    print("🧪 Starting AI Backtesting System Test Suite")
+    print("=" * 60)
+    
+    # Create test suite
+    suite = unittest.TestLoader().loadTestsFromTestCase(TestSystemIntegration)
+    
+    # Run tests with detailed output
+    runner = unittest.TextTestRunner(verbosity=2, stream=open(os.devnull, 'w'))
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)